project/src/test/java/com/google/sps/GetLabelCategoriesTest.java

```diff
@@ -0,0 +1,170 @@
+    Set<String> EVENT_LABELS = new HashSet<>(Arrays.asList("Soccer", "Robotics"));
+    Set<String> USER_INTERESTS = new HashSet<>(Arrays.asList("Basketball"));
+    Set<String> IRRELEVANT_LABELS = new HashSet<>(Arrays.asList("Cooking"));
+    Event event = TestUtils.newEventNoLabels().toBuilder().setLabels(EVENT_LABELS).build();
+    Event event2 = TestUtils.newEventNoLabels().toBuilder().setLabels(IRRELEVANT_LABELS).build();
+    User user = TestUtils.newUserNoInterestSkills().toBuilder()
+      .setInterests(USER_INTERESTS).build();
+    Set<Pair<Event, Integer>> expectedSimilarMatches = 
+      new HashSet<>(Arrays.asList(new Pair<Event, Integer>(event, /*common labels=*/ 1)));
+
+    ArrayList<Set<Pair<Event, Integer>>> matches = labelCategories.getEventRelevancy(
+      new HashSet<>(Arrays.asList(event, event2)), user);
+    Set<Pair<Event, Integer>> directMatches = matches.get(0);
+    Set<Pair<Event, Integer>> similarMatches = matches.get(1);
+
+    Assert.assertEquals(new HashSet<>(), directMatches);
+    Assert.assertEquals(expectedSimilarMatches, similarMatches);
+  }
+
+  @Test
+  /** Tests finding direct and similar matches between multiple events
+    * Relations: Basketball, Soccer, Swimming all sports; Card Games is subcategory of Games
+  */
+  public void directAndSimilarMatch() throws FileNotFoundException {
+    Set<String> USER_INTERESTS = new HashSet<>(Arrays.asList("Basketball", "Travel", "Games", "Robotics"));
+    Set<String> DIRECT_INDIRECT_MATCH_LABELS = new HashSet<>(Arrays.asList("Soccer", "Robotics"));
+    Set<String> SWIMMING_LABEL = new HashSet<>(Arrays.asList("Swimming"));
+    Set<String> MORE_DIRECT_INDIRECT_MATCH_LABELS = new HashSet<>(Arrays.asList("Cooking","Card Games", "Soccer" ));
+    Event event = TestUtils.newEventNoLabels().toBuilder().setLabels(DIRECT_INDIRECT_MATCH_LABELS).build();
+    Event event2 = TestUtils.newEventNoLabels().toBuilder().setLabels(MORE_DIRECT_INDIRECT_MATCH_LABELS).build();
+    Event event3 = TestUtils.newEventNoLabels().toBuilder().setLabels(SWIMMING_LABEL).build();
+    Event event4 = TestUtils.newEventNoLabels();
+    User user = TestUtils.newUserNoInterestSkills().toBuilder()
+      .setInterests(USER_INTERESTS).build();
+    Set<Pair<Event, Integer>> expectedDirectMatches = 
+      new HashSet<>(Arrays.asList(new Pair<Event, Integer>(event, /*common labels=*/ 1)));
+    Set<Pair<Event, Integer>> expectedSimilarMatches = 
+      new HashSet<>(Arrays.asList(new Pair<Event, Integer>(event, /*common labels=*/ 1),
+      new Pair<Event, Integer>(event2, /*common labels=*/ 2), new Pair<Event, Integer>(event3, /*common labels=*/ 1)));
+
+    ArrayList<Set<Pair<Event, Integer>>> matches = labelCategories.getEventRelevancy(
+      new HashSet<>(Arrays.asList(event, event2, event3, event4)), user);
+    Set<Pair<Event, Integer>> directMatches = matches.get(0);
+    Set<Pair<Event, Integer>> similarMatches = matches.get(1);
+
```

Comment:
I think you can inline these two lines since variables are only used once in assert.

Reply:
inlined w comments for all, done